### ISSUE TITLE
cannot enable hardware sync when deploying a machine (maas_instance resource)

### ISSUE THREAD

**Issue by pescobar** [maas/terraform-provider-maas#57]
Hi,

When deploying a new machine using terraform there is no option to enable hardware sync as you can do when using the webui (see screenshot)

![deploy-maas](https://user-images.githubusercontent.com/103797/231723304-119f9e23-7d17-4546-9f04-8309039d21d5.png)

Would it be possible to extend the `maas_instance` resource so it allows to enable hardware sync on deployment?

My test environment is MAAS 3.3.2 with terraform v1.4.2 and maas provider 1.1.0

**Comment by mastier** [comment]
It should be quite straight forward as this is covered via [MAAS api](https://maas.io/docs/api)

POST /MAAS/api/2.0/machines/{system_id}/op-deploy: Deploy a machine

- enable_hw_sync (boolean): Optional. If true, machine will be deployed with a small agent periodically pushing hardware data to detect any change in devices.

It would be somewhere here [resource_maas_instance.go#L83-L108](https://github.com/maas/terraform-provider-maas/blob/3729b9223661e9e4b11162a42054416fbb0b6fa5/maas/resource_maas_instance.go#L83-L108)
in the code

**Comment by mastier** [comment]
@pescobar I was not able to test it thoroughly, but you may compile it from my fork

Here's PR : https://github.com/maas/terraform-provider-maas/pull/59

**Comment by pescobar** [comment]
I tried this patch and the terraform provider accepted the new argument `enable_hw_sync = true` . I used these `deploy_params` in the `maas_instance` resource:

```
     deploy_params {
         distro_series = "jammy"
         user_data     = file("cloud-init/users.yml")
         enable_hw_sync = true
     }
```

But still the hardware sync is not enabled for the deployed `maas_instance`

I think interaction with the MAAS api is handled by [gomaasclient](https://github.com/maas/gomaasclient) and I have seen [these lines](https://github.com/maas/gomaasclient/blob/master/entity/machine.go#L157-L170) 

I have no experience with golang but my guess is that the gomaasclient also needs to be updated for this feature to work. Could it be?

**Comment by mastier** [comment]
Yes, of course. Good find! Could you please provide me a simple but full terraform example to test.
I might check if this is possible to be implemented easily.

**Comment by pescobar** [comment]
this is the terraform code I am testing (you will need to adapt it to your env)

Once terraform is working you should be able to do `terraform apply` to register the physical machine in maas and deploy it. Let me know if you want more details about terraform setup.

```
# maas_machine.node08:
resource "maas_machine" "node08" {
    architecture     = "amd64/generic"
    domain           = "maas"
    hostname         = "node08"
    pool             = "default"
    power_parameters = {
      cipher_suite_id = "3"
      k_g             =  ""
      mac_address     = "18:C0:4D:AA:AA:AA"
      power_address   = "192.168.34.34"
      power_boot_type = "efi"
      power_driver    = "LAN_2_0"
      power_pass      = "xxxxxxxxxx",
      power_user      = "yyyyyyyyyy"
      privilege_level = "ADMIN"

    }
    power_type       = "ipmi"
    pxe_mac_address  = "b4:96:91:BB:BB:BB"
    zone             = "default"

    timeouts {}
}

resource "maas_instance" "node08" {
    depends_on = [maas_machine.node08]
    allocate_params {
        hostname = maas_machine.node08.hostname
        pool     = "default"
        zone     = "default"
    }
    deploy_params {
        distro_series = "jammy"
        enable_hw_sync = true
    }
}
```

I am trying to patch the gomaasclient (see https://github.com/pescobar/gomaasclient/commit/bc754b7a10911700c8107306ea83870b7eb5f128) but I am probably doing it wrong, fist time I ready golang ;)
